VirtoCommerce.Platform.Web/Scripts/app/security/blades/account-resetPassword.js

```diff
@@ -1,27 +1,37 @@
-﻿angular.module('platformWebApp')
-.controller('platformWebApp.accountResetPasswordController', ['$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.accounts', function ($scope, bladeNavigationService, accounts) {
+angular.module('platformWebApp')
+.controller('platformWebApp.accountResetPasswordController', ['$q', '$scope', 'platformWebApp.bladeNavigationService', 'platformWebApp.accounts', 'platformWebApp.passwordValidationService', function ($q, $scope, bladeNavigationService, accounts, passwordValidationService) {
     var blade = $scope.blade;
 
     function initializeBlade() {
         blade.currentEntity = {
             newPassword: '',
-            newPassword2: ''
+            forcePasswordChange: true,
+            passwordIsValid: true,
+            minPasswordLength: 0,
+            errors: []
         };
         
         blade.isLoading = false;
     };
-    
-    $scope.saveChanges = function () {
-        if (blade.currentEntity.newPassword != blade.currentEntity.newPassword2) {
-            blade.error = 'Error: new passwords doesn\'t match!';
-            return;
-        }
 
+    $scope.validatePasswordAsync = function (value) {
+        var promise = passwordValidationService.validatePasswordAsync(value).then(
+            function(result) {
+                angular.extend(blade.currentEntity, result);
+
+                return result.passwordIsValid ? $q.resolve() : $q.reject();
```

Comment:
Why do you use $q.resolve() : $q.reject(); this? I think all can work without these calls. Just returns result.passwordIsValid should be enought.